lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -530,6 +533,88 @@ def internal_check_future_reserved_word(o, name)
     end
   end
 
+  def internal_check_file_scope(o, name, file)
+    return if file.nil?
+
+    file_scope = @file_scope_hash[file]
+    if file_scope.nil?
+      return if @file_scope_hash.key?(file)
+      file_scope = @file_scope_hash[file] = generate_file_scope file
```

Comment:
Since this is not some kind of DSL, I would prefer the call `generate_file_scope file` to be done with parentheses around `file`. Parentheses seems arbitrarily dropped as they are present almost everywhere else in calls with one argument.

Reply:
With the suggested renamed methods - the line would be
`file_ns = @file_to_namespace[file] = namespace_for_file(file)`